### PR TITLE
varnish: don't pass cookies to backend for load.php

### PR DESCRIPTION
**Pull request by RhinosF1**

### CODE REVIEW

**Comment by Universal-Omega** [comment]
There is no cookies set for load.php is there?

**Comment by Universal-Omega** [comment]
> `req.url !~ "^/w/load\.php"`
Is in the if statement, therefore the `else` is called for that, calling `mw_stash_cookie` which removes the cookie values it seems.

**Comment by RhinosF1** [comment]
> There is no cookies set for load.php is there?

Any cookie is set on the domain

> > `req.url !~ "^/w/load\.php"`
> 
> Is in the if statement, therefore the `else` is called for that, calling `mw_stash_cookie` which removes the cookie values it seems.

It's an or statement that includes session cookies causing cache to be bypassed. This allows it to be cached safely.

**Comment by Universal-Omega** [comment]
OK well what exact is the point of doing this, any real benefit then?

**Comment by RhinosF1** [comment]
> OK well what exact is the point of doing this, any real benefit then? I may not be fully understanding though.

To serve cached resources from load.php to logged in users

**Comment by Universal-Omega** [comment]
OK sounds good then. I understand what you mean now, I was just not understanding before. Thanks!

**Comment by RhinosF1** [comment]
Should we do this?

File: modules/varnish/templates/default.vcl [hunk 1]
```diff
@@ -275,6 +275,10 @@ sub mw_stash_cookie {
 }
 
 sub mw_evaluate_cookie {
+        if (req.url ~ "^/w/load\.php") {
+           unset req.http.Cookie; # load.php URLs should be specfic and sessions disabled, we can ignore cookies
```
Comment (JohnFLewis) [inline modules/varnish/templates/default.vcl:279]: can this not use mw_stash_cookie like below? Also is this not triggered by the below?

Comment (Universal-Omega) [reply modules/varnish/templates/default.vcl:279]: That's what I thought also but was told above that it wasn't.

**Comment by JohnFLewis** [comment]
I am looking at overhauling major parts of caching anyway, so closing for now.

**Pull request closed without merging**